app/uk/gov/hmrc/economiccrimelevyregistration/controllers/BaseController.scala

```diff
@@ -17,27 +17,23 @@
 package uk.gov.hmrc.economiccrimelevyregistration.controllers
 
 import cats.data.EitherT
-import play.api.Logger
 import play.api.http.HeaderNames.CACHE_CONTROL
 import play.api.i18n.{I18nSupport, Messages}
 import play.api.mvc.Results.{InternalServerError, Redirect}
 import play.api.mvc.{Request, RequestHeader, Result, Results}
 import play.twirl.api.Html
 import uk.gov.hmrc.economiccrimelevyregistration.models.{Mode, Registration}
-import uk.gov.hmrc.economiccrimelevyregistration.models.errors.{DataRetrievalError, ErrorCode, ResponseError}
+import uk.gov.hmrc.economiccrimelevyregistration.models.errors.{ErrorCode, ResponseError}
 import uk.gov.hmrc.economiccrimelevyregistration.navigation.PageNavigator
 import uk.gov.hmrc.economiccrimelevyregistration.views.html.ErrorTemplate
 
 import scala.concurrent.{ExecutionContext, Future}
 
 trait BaseController extends I18nSupport {
 
-  def getValue[T](option: Option[T]): EitherT[Future, DataRetrievalError, T] =
+  def valueOrError[T](value: Option[T], message: String = "Missing value") =
```

Comment:
Message is not useful from the investigators point of view. Can we pass in the value name?